components/slack/actions/send-message-public-channel/send-message-public-channel.js

```diff
@@ -47,11 +47,48 @@ module.exports = {
       ],
       description: "Optionally provide an image URL to use as the bot icon for this message.",
     },
+    include_sent_via_pipedream_flag: {
+      type: "boolean",
+      optional: true,
+      default: true,
+      label: "Include link to workflow",
+      description: "Defaults to `true`, includes a link to the workflow at the end of your Slack message.",
+    },
   },
   async run() {
+    let link = `https://pipedream.com/@/${process.env.PIPEDREAM_WORKFLOW_ID}`;
+    link += `/inspect/${process.env.PIPEDREAM_TRACE_ID}`;
+    link += "?origin=action";
+    link += "&a=slack";
+
+    const sentViaPipedreamText =
+      {
+        "type": "context",
+        "elements": [
+          {
+            "type": "mrkdwn",
+            "text": `Sent via <${link}|Pipedream>`,
+          },
+        ],
+      };
+
+    if (this.include_sent_via_pipedream_flag == true) {
+      this.blocks = [
+        {
+          "type": "section",
+          "text": {
+            "type": "mrkdwn",
+            "text": this.text,
```

Comment:
this chokes when I send an object via the text prop (only when `include_sent_via_pipedream_flag` == `true`)